tests/test_api.py

```diff
@@ -157,6 +177,27 @@ def test_word_confidences(self):
         mapped_confidences = self._api.MapWordConfidences()
         self.assertEqual([v[0] for v in mapped_confidences], words)
         self.assertEqual([v[1] for v in mapped_confidences], confidences)
+        
+    def test_LSTM_choices(self):
+        if _TESSERACT_VERSION >= 0x4000000:
+            """Test GetBestLSTMSymbolChoices."""
+            self._api.SetVariable("lstm_choice_mode", "2")
+            self._api.SetImageFile(self._image_file)
+            self._api.Recognize()
+            LSTM_choices = self._api.GetBestLSTMSymbolChoices()
+            words = self._api.AllWords()
+            self.assertEqual(len(words), len(LSTM_choices))
+            
+            for choice, word in zip(LSTM_choices, words):
+                chosen_word = ""
+                for timestep in choice:
+                    for alternative in timestep:
+                        self.assertGreaterEqual(alternative[1], 0.0)
+                        self.assertLessEqual(alternative[1], 2.0)
```

Comment:
@noahmetzger, what was the reason for the upper bound 2.0?

The test fails with the current code:

    FAIL: test_LSTM_choices (tests.test_api.TestTessBaseApi)
    Test GetBestLSTMSymbolChoices.
    ----------------------------------------------------------------------
    Traceback (most recent call last):
      File "/media/home/debian/src/github/OCR-D/ocrd_all/tesserocr/tests/test_api.py", line 196, in test_LSTM_choices
        self.assertLessEqual(alternative[1], 2.0)
    AssertionError: 46.24480056762695 not less than or equal to 2.0

Reply:
I think this was part of an older version where the probabilities for alternatives have been between 0 and 1.0 according to their percentage. 2.0 was a special marker for the value which was chosen by tesseract for the final result. But this isn't in the code of tesseract anymore.